app/src/main/java/com/mobile/tandil/kotlinbaseproject/activity/ParkingStayActivity.kt

```diff
@@ -0,0 +1,87 @@
+import com.mobile.tandil.kotlinbaseproject.fragment.DatePickerDialogFragment
+import com.mobile.tandil.kotlinbaseproject.fragment.TimePickerDialogFragment
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.ParkingStayViewModel
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.ParkingStayViewModel.CheckState
+import com.mobile.tandil.kotlinbaseproject.mvvm.viewmodel.ParkingStayViewModel.ParkingStayData
+
+class ParkingStayActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityParkingStayBinding
+    private val viewModel = ParkingStayViewModel()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityParkingStayBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        viewModel.getValue().observe({ lifecycle }, ::updateUI)
+
+        setListeners()
+    }
+
+    private fun setListeners() {
+        binding.startDateTimeButton.setOnClickListener {
+            viewModel.showStartTimePickerFragment()
+            viewModel.showStartDatePickerFragment()
+        }
+        binding.endDateTimeButton.setOnClickListener {
+            viewModel.showEndTimePickerFragment()
+            viewModel.showEndDatePickerFragment()
+        }
+        binding.confirmButton.setOnClickListener {
+            viewModel.closeActivity()
+        }
+        binding.cancelButton.setOnClickListener {
+            viewModel.closeActivity()
+        }
+    }
+
+    private fun updateUI(data: ParkingStayData) {
+        when (data.state) {
+            CheckState.SHOW_START_DATE_PICKER_FRAGMENT -> {
+                val datePicker = DatePickerDialogFragment { day, month, year -> onStartDateSelected(day, month, year) }
+                datePicker.show(supportFragmentManager, DatePickerDialogFragment.DATE_PICKER_DIALOG)
+            }
+            CheckState.SHOW_START_TIME_PICKER_FRAGMENT -> {
+                val timePicker = TimePickerDialogFragment { onStartTimeSelected(it) }
+                timePicker.show(supportFragmentManager, TimePickerDialogFragment.TIME_PICKER_DIALOG)
+            }
+            CheckState.SHOW_END_DATE_PICKER_FRAGMENT -> {
+                val datePicker = DatePickerDialogFragment { day, month, year -> onEndDateSelected(day, month, year) }
+                datePicker.show(supportFragmentManager, DatePickerDialogFragment.DATE_PICKER_DIALOG)
+            }
+            CheckState.SHOW_END_TIME_PICKER_FRAGMENT -> {
+                val timePicker = TimePickerDialogFragment { onEndTimeSelected(it) }
+                timePicker.show(supportFragmentManager, TimePickerDialogFragment.TIME_PICKER_DIALOG)
+            }
+            CheckState.CLOSE_ACTIVITY -> {
+                val intent = Intent(this, MainActivity::class.java)
+                startActivity(intent)
+            }
+        }
+    }
+
+    private fun onStartDateSelected(day: Int, month: Int, year: Int) {
+        val pickedDay = "$day/$month/$year"
```

Comment:
if you are not using this value anywhere else, please remove it

Reply:
done